cmd/yace/main.go

```diff
@@ -100,29 +100,34 @@ func main() {
 			for {
 				t0 := time.Now()
 				newRegistry := prometheus.NewRegistry()
-				endtime := exporter.UpdateMetrics(config, newRegistry, now, *metricsPerQuery, *fips, *floatingTimeWindow, *labelsSnakeCase, staticJobSemaphore, discoveryJobSemaphore)
-				now = endtime
-				log.Debug("Metrics scraped.")
-				registry = newRegistry
-				t1 := time.Now()
-				processingtime := t1.Sub(t0)
-				processingtimeTotal = processingtimeTotal + processingtime
-				if processingtimeTotal.Seconds() > 60.0 {
-					sleepinterval := *scrapingInterval - int(processingtimeTotal.Seconds())
-					//reset processingtimeTotal
-					processingtimeTotal = 0
-					if sleepinterval <= 0 {
-						//TBD use cases is when metrics like EC2 and EBS take more scrapping interval like 6 to 7 minutes to finish
-						log.Debug("Unable to sleep since we lagging behind please try adjusting your scrape interval or running this instance with less number of metrics")
-						continue
+				endtime, errors := exporter.UpdateMetrics(config, newRegistry, now, *metricsPerQuery, *fips, *floatingTimeWindow, *labelsSnakeCase, staticJobSemaphore, discoveryJobSemaphore)
+				if len(errors) > 0 {
+					//TODO does this even look nice?
```

Comment:
Might add the total number of errors to the log message?

Also, how's the slice of errors printed? Is each error printed individually? Otherwise maybe could have each error printed as a separate log line?